uitools/import/Esri/ArcGISRuntime/Toolkit/FloorFilter.qml

```diff
@@ -0,0 +1,599 @@
+                                onClicked: {
+                                    controller.setSelectedLevelId(model.modelId);
+                                }
+                            }
+                        }
+                    }
+                }
+
+                ToolSeparator {
+                    visible: repeater.count !== 0
+                    Layout.fillWidth: true
+                    orientation: Qt.Horizontal
+                }
+
+                // not visible when floorFilter is shown and not visible if the children has no text to be shown.
+                ToolButton {
+                    id: itemSelectedButton
+                    objectName: "itemSelectedButton"
+                    visible: closer.checked && text !== ""
+                    Layout.fillWidth: true
+                    // always checked
+                    checkable: false
+                    checked: true
+                    text: controller.selectedLevel ? (internal.collapsedIcons ? controller.selectedLevel.shortName : controller.selectedLevel.longName) : ""
+                    flat: true
+                    display: AbstractButton.TextOnly
+                    onClicked: {
+                        closer.checked = false;
+                    }
+                }
+
+                ToolSeparator {
+                    visible: itemSelectedButton.visible
+                    Layout.fillWidth: true
+                    orientation: Qt.Horizontal
+                }
+
+                ToolButton {
+                    id: buildingMenuButton
+                    objectName: "buildingMenuButton"
+                    checkable: true
+                    Layout.fillWidth: true
+                    Layout.alignment: Qt.AlignLeft
+                    visible: !hideSiteFacilityButton
+                    icon.source: "images/organization.svg"
+                    text: "Browse"
+                    flat: true
+                    display: internal.collapsedIcons ? AbstractButton.IconOnly : AbstractButton.TextBesideIcon
+                }
+            }
+        }
+
+        Pane {
+            padding: 0
+            visible: buildingMenuButton.checked
+            GridLayout {
+                flow: GridLayout.TopToBottom
+                Layout.alignment: Qt.AlignBottom
+                rows: 6
+                columnSpacing: 0
+                rowSpacing: 0
+                Button {
+                    objectName: "backToSite"
+                    Layout.margins: 0
+                    Layout.fillHeight: true
+                    Layout.rowSpan: 2
+                    Layout.alignment: Qt.AlignHCenter
+                    // setting width to 0 and not setting visibility to false, otherwise it would break the rows layout by having a missing row.
+                    Layout.preferredWidth: internal.currentVisibileListView
+                                           === FloorFilter.VisibleListView.Facility && controller.sites.count > 0 ? 32 : 0
```

Comment:
32 is the default value I find looks good in case the currentView is facility and there are sites. Otherwise, hide the button by setting its height to 0